dist/jquery.easy-autocomplete.js

```diff
@@ -418,23 +418,23 @@ var EasyAutocomplete = (function (scope) {
 			return listBuilder;
 		};
 
-		this.updateCategories = function (listBuilder, data) {
+    this.updateCategories = function(listBuilder, data) {
+      if (configuration.get("categoriesAssigned")) {
+        listBuilder = [];
 
-			if (configuration.get('categoriesAssigned')) {
+        $.each(data, function( index, value ) {
+          var builder = convertToListBuilder({
+            listLocation: index,
+            maxNumberOfElements: 10,
+            header: index.replace(/[_\s]+/g, ' ')
+          }, data);
 
-				listBuilder = [];
+          listBuilder.push(builder);
+        });
+      }
 
-				for (var i = 0; i < configuration.get("categories").length; i += 1) {
-
-					var builder = convertToListBuilder(configuration.get('categories')[i], data);
-
-					listBuilder.push(builder);
-				}
-
-			}
-
-			return listBuilder;
-		};
```

Comment:
instead of removing it, check if `config. categories` exist, use it; otherwise, build it dynamically.